packages/story-editor/src/components/panels/design/videoOptions/videoOptions.js

```diff
@@ -88,24 +87,52 @@ const HelperText = styled(Text).attrs({
 
 function VideoOptionsPanel({ selectedElements, pushUpdate }) {
   const { isTranscodingEnabled } = useFFmpeg();
-  const { muteExistingVideo } = useLocalMedia((state) => ({
-    muteExistingVideo: state.actions.muteExistingVideo,
-  }));
+  const {
+    muteExistingVideo,
+    isResourceTrimming,
+    isResourceMuting,
+    isResourceProcessing,
+    isResourceProcessingById,
+  } = useLocalMedia(
+    ({
+      state: {
+        isResourceProcessing,
+        isResourceProcessingById,
+        isResourceMuting,
+        isResourceTrimming,
+      },
+      actions: { muteExistingVideo },
+    }) => ({
+      isResourceProcessing,
+      isResourceProcessingById,
+      isResourceMuting,
+      isResourceTrimming,
+      muteExistingVideo,
+    })
+  );
   const resource = getCommonValue(selectedElements, 'resource');
-  const { isMuting, isMuted, isTrimming } = resource;
+  const { isMuted, id: resourceId = 0 } = resource;
+  const isTrimming = isResourceTrimming(resourceId);
+  const isMuting = isResourceMuting(resourceId);
   const loop = getCommonValue(selectedElements, 'loop');
   const isSingleElement = selectedElements.length === 1;
 
   const handleMuteVideo = useCallback(() => {
     muteExistingVideo({ resource });
   }, [resource, muteExistingVideo]);
 
-  const shouldDisableVideoActions = !canTranscodeResource(resource);
+  const shouldDisableVideoActions =
+    isResourceProcessing(resourceId) ||
+    isResourceProcessingById(resourceId) ||
+    resource.isExternal;
 
   const shouldDisplayMuteButton =
     isTranscodingEnabled &&
     isSingleElement &&
-    ((!isMuted && canTranscodeResource(resource)) || isMuting);
+    ((!isMuted &&
+      !isResourceProcessing(resourceId) &&
+      !isResourceProcessingById(resourceId)) ||
+      isMuting);
```

Comment:
Q: Doesn't `isResourceProcessing` already include `isMuting` as well? Meaning that doesn't `!isResourceProcessing` already mean it's not muting either?

Reply:
This has to do with naming; a reason why we need to change it.

Currently, the `byId` function means "check if this current media item is being processed", where es the one without the suffix means "check if a new version of this resource is being processed".

When muting an existing video in a story, you are at the same time "processing" that existing video, but also uploading a new one (with its own unique ID), which is being displayed in the media library.

Now, we want two be able to do two things:

1. show a progress bar for the newly uploading muted video in the library
2. Show the "Muting..." button for the old, existing video in the story